utils/common/src/main/kotlin/Extensions.kt

```diff
@@ -277,7 +277,7 @@ fun <K, V : Collection<T>, T> Map<K, V>.zipWithCollections(other: Map<K, V>): Ma
     zip(other) { a, b ->
         when {
             // When iterating over the combined key set, not both values can be null.
-            a == null -> b!!
+            a == null -> checkNotNull(b)
```

Comment:
Java 14 improved the `NullPointerExceptions` to display the actual variable name which was null causing this exception.
`checkNotNull()` only shows `Required value was null.`.

I'm not sure if this Java 14 feature would also work with NPEs from Kotlin, and if it would be more helpful. But it might be one thing to consider.

Reply:
Aside from that, when looking at this commit, if these are the only places where it had to be changed, I would also vote for keeping this the rule to disallow `!!`.

Reply:
Additionally, this rule seems not to disallow _every_ usage of `!!`.

Reply:
> I'm not sure if this Java 14 feature would also work with NPEs from Kotlin, and if it would be more helpful.

That's a nice Java feature indeed, and I believe it would work with any JVM language; however, for the time being we're probably stuck with Java 11 anyway, also see https://github.com/oss-review-toolkit/ort/pull/4948 😞 

> Aside from that, when looking at this commit, if these are the only places where it had to be changed

To be fair, there might be might more places (in non-main source sets) that would need to be changed when keeping this rule enabled *and* using Detekt with type resolution. I'm not yet done with looking at all the code, but I'll probably stop here for now with this preparation.